### DISCUSSION TITLE
Add classes property to node/edge in cytoscape_data

### DISCUSSION THREAD

**Discussion by slai-nick** [networkx/networkx#5927] [Ideas]
Cytoscape JSON format has an optional `classes` property which can be used to style node/edges by group.
Currently there is no option to generate this property with `cytoscape_data` and the user has to do add it manually after generation for each desired edge/node.

It would be great to have a `classes` parameter, maybe one for node and one for edges, that takes a mapping: node/edge -> class or list of class.

That would make it easier and save some performance as I imagine this can easily be done during the generation.

**Comment by dschult** [comment]
You can store a node/edge attribute on the graph G named `classes` and put the information in there. Then the `cytoscape_data` function will add that to the dict info that is included.
```python
G=nx.path_graph(3)
G.nodes[0]["classes"] = "bar"
G[0][1]["class"]="foo"
nx.cytoscape_data(G)
```
with result
```
{'data': [],
 'directed': False,
 'multigraph': False,
 'elements': {'nodes': [{'data': {'classes': "bar", 'id': '0', 'value': 0, 'name': '0'}},
   {'data': {'id': '1', 'value': 1, 'name': '1'}},
   {'data': {'id': '2', 'value': 2, 'name': '2'}}],
  'edges': [{'data': {'classes': 'foo', 'source': 0, 'target': 1}},
   {'data': {'source': 1, 'target': 2}}]}}
```
